### PR TITLE
Feat/bertopic

### PR DESCRIPTION
**Pull request by erob123**
Going over in-person today!

### CODE REVIEW

File: Pipfile
Comment (emiliecowen) [inline Pipfile:1]: (discussed with @erob123 on 20230426) current pipenv package management is set up for arm64. intel workaround is to pip install -r requirements.txt; work with eric to capture new package needs in the future.

File: README.md [hunk 1]
```diff
@@ -17,19 +17,49 @@ conda activate c3po-os-api
 
 7. Install dependencies and dev dependencies locally from the `Pipfile` by running `pipenv install --dev`.
```
Comment (emiliecowen) [inline README.md:18]: append: (Current workaround for intel platforms: `pip install -r requirements.txt`, pip install additional packages needed for new development, notify Eric.)

File: README.md [hunk 2]
```diff
@@ -17,19 +17,49 @@ conda activate c3po-os-api
...
 
-8. Start the app `pipenv run uvicorn app.api:versioned_app`.
-
-9. Post a request in a separate shell!
-```bash
-curl -X POST -H "Content-Type: application/json" \
-    -d '{"text": "A fun little note."}' \
-    http://localhost:8000/predict
+8.  Create a `secrets.env` file and ensure it is on the `.gitignore`.  Add the following for local dev:
```
Comment (emiliecowen) [inline README.md:20]: replace "Create a `secrets.env`" with "In c3po-model-server/app/core/env_var, create a `secrets.env`"

File: app/aimodels/bertopic/routers/documents.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from sqlalchemy.orm import Session
+from .. import crud
+from ..models.document import DocumentModel
+from app.core.errors import HTTPValidationError, ValidationError
+
+router = APIRouter(
+    prefix="/documents"
+)
+
+@router.post(
```
Comment (emiliecowen) [inline app/aimodels/bertopic/routers/documents.py:16]: issue entering document time, entered https://github.com/MIT-AI-Accelerator/c3po-model-server/issues/7 to address later

File: README.md [hunk 4]
```diff
@@ -17,19 +17,49 @@ conda activate c3po-os-api
...
+13. Keeping your docker containers running, start the app in a new terminal (activate your conda env first) with `pipenv run uvicorn app.main:versioned_app --reload`.
+
+14. Open `localhost:8000/v1/docs` and start interacting with swagger!
+
+15. Run tests and get coverage with `pytest --cov=app`
+
+16.  You can shut down and your db / minio data will persist via docker volumes.
+
 
 # Notes
```
Comment (emiliecowen) [inline README.md:52]: having trouble configuring pgadmin to see the tables were created properly

File: scripts/prestart.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+# run from root of project
+
+# Let the DB start
+# python /app/app/backend_pre_start.py
+
+# Run migrations
+# alembic upgrade head
+
+# Create initial data in DB
+python3 -m app.initial_data
```
Comment (emiliecowen) [inline scripts/prestart.sh:11]: ModuleNotFound error for python3 (sqlalchemy.orm). The env setup procedure sets up modules for 'python', but 'python3' is still pointing to some other version installed outside of the c3po environment. Recommend changing this line to python -m app.initial_data

File: Pipfile
Comment (emiliecowen) [reply Pipfile:1]: document here: https://app.slack.com/client/T0466M3J485/C055ZTNB20Y/thread/C04941YSSHM-1680630238.741949

File: scripts/prestart.sh (hunk 5 above)
Comment (emiliecowen) [reply scripts/prestart.sh:11]: also having trouble confirming the initialization works via pgadmin (left a related note on the README)

File: app/aimodels/bertopic/routers/bertopic_embedding_pretrained.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+from aiofiles import open as open_aio
+from minio import Minio
+
+
+router = APIRouter(
+    prefix="/bertopic-embedding-pretrained"
+)
+
+
+@router.post(
```
Comment (emiliecowen) [inline app/aimodels/bertopic/routers/bertopic_embedding_pretrained.py:21]: created story to pickle, save, and return sha256 for sentence_models prior to upload

**Review by emiliecowen — CHANGES_REQUESTED** [review]
@erob123 there are a few comments that should be resolved before the merge; please let me know if you have any questions.

File: README.md (hunk 1 above)
Comment (erob123) [reply README.md:18]: Added clarification in readme

File: README.md (hunk 2 above)
Comment (erob123) [reply README.md]: Fixed

File: app/aimodels/bertopic/routers/documents.py (hunk 3 above)
Comment (erob123) [reply app/aimodels/bertopic/routers/documents.py:16]: agreed

File: README.md (hunk 4 above)
Comment (erob123) [reply README.md:53]: discuss today

File: scripts/prestart.sh (hunk 5 above)
Comment (erob123) [reply scripts/prestart.sh]: fixed python vs python3...will discuss pgadmin today

File: app/aimodels/bertopic/routers/train.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+class TrainModelRequest(BaseModel):
+    bertopic_embedding_pretrained_id: UUID4
+    document_ids: list[UUID4] = []
+
+
+@router.post(
+    "/train",
+    response_model=Union[BertopicTrained, HTTPValidationError],
+    responses={'422': {'model': HTTPValidationError}},
```
[the remark this answers is not in the document]
Comment (erob123) [reply app/aimodels/bertopic/routers/train.py:29]: agreed

File: app/topics/insights/routers/team_trending.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+class VisualizationRequest(BaseModel):
+    team: str
+
+
+class VisualizationResponse(BaseModel):
+    plot_params: dict
+
+
+@router.post("/visualization/", response_model=VisualizationResponse)
```
[the remark this answers is not in the document]
Comment (erob123) [reply app/topics/insights/routers/team_trending.py:20]: agreed

**Review by erob123 — COMMENTED** [review]
All comments addressed

**Review by emiliecowen — APPROVED** [review]
confirmed pytests pass and pre_init runs without error

**Pull request merged**